### PR TITLE
Switch to Pytest's terminal reporter

### PR DESCRIPTION
**Pull request by s0undt3ch**
Additionally update pre-commit hook versions

### CODE REVIEW

File: pytest_timeout.py [hunk 1]
```diff
@@ -158,9 +157,11 @@ def pytest_runtest_protocol(item):
...
     if is_timeout and settings.func_only is False:
         hooks.pytest_timeout_set_timer(item=item, settings=settings)
-    yield
-    if is_timeout and settings.func_only is False:
-        hooks.pytest_timeout_cancel_timer(item=item)
+    try:
+        yield
+    finally:
+        if is_timeout and settings.func_only is False:
+            hooks.pytest_timeout_cancel_timer(item=item)
```
Comment (flub) [inline pytest_timeout.py:164]: this seems like quite an important bugfix.  could you update the changelog as well?

File: test_pytest_timeout.py [hunk 2]
```diff
@@ -16,92 +16,83 @@
...
-def test_header(testdir):
-    testdir.makepyfile(
+def test_header(pytester):
+    pytester.makepyfile(
         """
         def test_x(): pass
     """
     )
-    result = testdir.runpytest("--timeout=1")
+    result = pytester.runpytest_subprocess("--timeout=0.5")
```
Comment (flub) [inline test_pytest_timeout.py:25]: I don't remember exactly, but I think this used to use fractional seconds to speed up the tests but did get reverted to 1s.  Probably for flakyness, maybe some other reasons.  Could you leave this to 1s please?

File: test_pytest_timeout.py [hunk 3]
```diff
@@ -551,8 +544,9 @@ def custom_trace(*args):
     assert pytest_timeout.is_debugging(custom_trace)
 
 
-def test_not_main_thread(testdir):
-    testdir.makepyfile(
+def test_not_main_thread(pytester):
+    pytest.skip("The 'pytest_timeout.timeout_setup' function no longer exists")
```
Comment (flub) [inline test_pytest_timeout.py:548]: i'm confused by this, why is there still a test for it then?

**Review by flub — COMMENTED** [review]
Thanks for the updates!  Looks good mostly, I'd like to request two things though:

- Keep the timeouts in the tests to 1s
- Add changelog entries for all the changes.

File: pytest_timeout.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply pytest_timeout.py:164]: With old style hook wrappers it's incorrect to add try/finally

Comment (s0undt3ch) [reply pytest_timeout.py:164]: Oh?!

Does the code after the `yield` always run then?

Got a link to something I could read? I have several plugins which might need to be updated...

What's the consequence?

File: test_pytest_timeout.py (hunk 2 above)
Comment (s0undt3ch) [reply test_pytest_timeout.py:25]: Sure

File: test_pytest_timeout.py (hunk 3 above)
Comment (s0undt3ch) [reply test_pytest_timeout.py:548]: This was broken in [dd9d6080763c559de1ff5b15deb9c27559dd3431](https://github.com/pytest-dev/pytest-timeout/commit/dd9d6080763c559de1ff5b15deb9c27559dd3431#diff-c1e97e79a9e00c3edb7cbb38156ac2c9567c2ce7101355cda025f38dc97d8009L190) and the test was broad enough that it kept passing.

**Comment by s0undt3ch** [comment]
> Thanks for the updates! Looks good mostly, I'd like to request two things though:
> 
>     * Keep the timeouts in the tests to 1s
> 
>     * Add changelog entries for all the changes.

Done.

File: pytest_timeout.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply pytest_timeout.py]: For legacy hook wrappers one needs to check the exception attribute if one needs to handle them

Moder hookwrappers Work like expected

**Review by flub — APPROVED** [review]
Thanks!  I guess maybe it's time to figure out how to do another release sometime soon.

**Comment by s0undt3ch** [comment]
> Thanks! I guess maybe it's time to figure out how to do another release sometime soon.

I'd love that!
I have a PR waiting for it 😁

**Pull request merged**